src/ray/util/thread_checker.cc

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/ray/util/thread_checker.h"
+
+namespace ray {
+
+bool ThreadChecker::ok() const {
```

Comment:
The overall runtime cost is ignorably small -- I'm pretty confident I could squeeze these CPU cycles from elsewhere.

If developers are really concerns about it (which means they care about nano seconds latency), they could use `RAY_DCHECK` to ignore at release.